ts/src/config.ts

```diff
@@ -115,16 +115,36 @@ export interface Config extends GoogleAuthOptions {
   // For testing with startLocal() only.
   localProfilingPeriodMillis?: number;
 
-
   // Debugging information for startLocal will be recorded every
   // localLogPeriodMillis milliseconds.
   // For testing with startLocal() only.
   localLogPeriodMillis?: number;
 
-
   // Duration of time profiles collected when using startLocal().
   // For testing with startLocal() only.
   localTimeDurationMillis?: number;
+
+  // List of directories recursively searched for *.js.map files. Defaults to
+  // process.cwd().
+  //
+  // The profiler uses these files to re-map the source file paths in the
+  // profiles. The most common use case of having a source map is an application
+  // or dependency written in TypeScript: the source file paths that
+  // the profiler observes in the profiling data are the transpiled *.js files;
+  // to attribute the data back to the TypeScript source that the developer
+  // wrote source map needs to be generated, distribute and used.
+  //
+  // The source map of the application typically resides directly in
+  // process.cwd() and the source maps of dependencies can normally be found
+  // via recursive search in process.cwd()/node_modules so the default value
+  // should work well pretty much always.
+  sourceMapSearchPath?: string[];
+
+  // Generally should not be set.
+  // Can be used to add a source map not within a directory any of the
+  // directories specified by sourceMapSearchPath, by specifying the absolute
+  // path to the source map in this array.
+  sourceMapPaths?: string[];
```

Comment:
My previous comment suggested (probably in a way that was not clear enough) that this configuration option is dropped. Is there a reason you decided to keep it? I am fine to keep it if you think there is a usage model for it that is important for most Node.js developers, but then clarify that in the doc string - currently the documentation for this option is obscure in terms of why it's important to have this option and is more of "nice to have though we don't really know why and when" knob.

Reply:
Updated this so this field is only in ProfilerConfig (the config passed to the Profiler class), not in Config (the config a user passes to the `start` function).